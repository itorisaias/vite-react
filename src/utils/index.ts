import { createFormAction } from "remix-forms";
import { redirect, json } from "react-router-dom";

export const formAction = createFormAction({ redirect, json });
