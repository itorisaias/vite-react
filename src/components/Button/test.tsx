import { describe, expect, it } from "vitest";
import { Button } from ".";
import { render } from "~test/utils";

describe("<Button />", () => {
  it("should render the medium size by default", () => {
    const { getByRole, debug } = render(<Button>Buy now</Button>);

    const btn = getByRole("button", { name: /buy now/i });

    expect(btn).toBeInTheDocument();
    expect(btn).toHaveStyle({
      height: "4rem",
      padding: "0.8rem 3.2rem",
      "font-size": "1.4rem",
    });
  });
});
