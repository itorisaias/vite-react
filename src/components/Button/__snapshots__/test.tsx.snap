// Vitest Snapshot v1

exports[`<Button /> > should render the medium size by default 1`] = `
<button
  class="sc-bcXHqe bFNpZh"
>
  <span>
    Buy now
  </span>
</button>
`;
